ansible_runner/streaming.py

```diff
@@ -0,0 +1,216 @@
+import codecs
+import io
+import json
+import os
+import stat
+import sys
+import tempfile
+import uuid
+import zipfile
+
+import ansible_runner
+import ansible_runner.plugins
+from ansible_runner import utils
+
+
+class UUIDEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, uuid.UUID):
+            return obj.hex
+        return json.JSONEncoder.default(self, obj)
+
+
+# List of kwargs options to the run method that should be sent to the remote executor.
+remote_run_options = (
+    'forks',
+    'host_pattern',
+    'ident',
+    'ignore_logging',
+    'inventory',
+    'limit',
+    'module',
+    'module_args',
+    'omit_event_data',
+    'only_failed_event_data',
+    'playbook',
+    'verbosity',
+)
+
+
+class Transmitter(object):
+    def __init__(self, _output=None, **kwargs):
+        if _output is None:
+            _output = sys.stdout.buffer
+        self._output = _output
+        self.kwargs = kwargs
+
+        self.status = "unstarted"
+        self.rc = None
+
+    def run(self):
+        remote_options = {key: value for key, value in self.kwargs.items() if key in remote_run_options}
+
+        private_data_dir = self.kwargs.get('private_data_dir', None)
+        self._output.write(
+            utils.stream_dir(private_data_dir, kwargs=json.dumps(remote_options, cls=UUIDEncoder)))
+        self._output.write(json.dumps({'eof': True}).encode('utf-8'))
+        self._output.write(b'\n')
+        self._output.flush()
+
+        return self.status, self.rc
+
+
+class Worker(object):
+    def __init__(self, _input=None, _output=None, **kwargs):
+        if _input is None:
+            _input = sys.stdin.buffer
+        if _output is None:
+            _output = sys.stdout.buffer
```

Comment:
The worker is also going to be writing to `sys.stdout` in other places like these:

https://github.com/ansible/ansible-runner/blob/e6020b7d33627a5d6656dca64d7ae09dd95cc75c/ansible_runner/utils.py#L288

Those will be non-json formatted lines... does `ansible-runner process` not get those? Does it get those lines and just pass over them?

Reply:
We do need to protect against this.  I'm not entirely clear how I'm going to do that yet.

Reply:
I ran your example and I can't find any stray output in `/tmp/output2`. I don't know why.

Reply:
`Worker` unconditionally gets the 'quiet' parameter set.  Hopefully everything that would leak output to stdout respects that parameter, but so far it looks like it does.